wandb/apis/public/api.py

```diff
@@ -1158,28 +1196,54 @@ def artifact(self, name, type=None):
         """
         if name is None:
             raise ValueError("You must specify name= to fetch an artifact.")
-        entity, project, artifact_name = self._parse_artifact_path(name)
+        org, entity, project, artifact_name = self._parse_artifact_path(name)
 
-        organization = ""
-        # If its an Registry artifact, the entity is an org instead
+        # If its an Registry artifact(under org entities), fetch org entity for user
         if is_artifact_registry_project(project):
-            # Update `organization` only if an organization name was provided,
-            # otherwise use the default that you already set above.
-            try:
-                organization, _, _ = name.split("/")
-            except ValueError:
-                organization = ""
-            # set entity to match the settings since in above code it was potentially set to an org
-            entity = self.settings["entity"] or self.default_entity
+            entity = self._fetch_org_entity_for_registry(org, entity)
+
         artifact = wandb.Artifact._from_name(
-            entity, project, artifact_name, self.client, organization
+            entity, project, artifact_name, self.client
         )
         if type is not None and artifact.type != type:
             raise ValueError(
                 f"type {type} specified but this artifact is of type {artifact.type}"
             )
         return artifact
 
+    def _fetch_org_entity_for_registry(self, organization: str, entity: str) -> str:
+        """Returns the org entity of an entity and validates the organization passed in is in the same org as the entity.
+
+        Arguments:
+            organization: (str) Organization or org entity name, possibly an empty string.
+            entity: (str) Entity of the Registry, used to fetch the org entity as well as
+                validate the organization passed in belongs to the same org as the org entity.
+
+        Returns:
+            org_entity (str) The org entity of the entity.
+        """
+        try:
+            org_entity = InternalApi()._resolve_org_entity_name(entity, organization)
```

Comment:
If its ok to use the InternalApi() method `resolve_org_entity_name` anyway, then why is there a re-implementation of the fetch method. Why can't you use the one in the InternalApi()? 

That'll also allow you to avoid parsing and passing in the org name everywhere, right?

I might be completely off base here, so lmk if a conversation offline will be better